src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -71,29 +80,81 @@ class PeerActor(nodeInfo: NodeInfo) extends Actor with ActorLogging {
+    case MessageReceived(status: Status) =>
+      timeout.cancel()
+      log.info("Peer {} returned status ({})", peerId, status)
+      rlpxConnection ! RLPxConnectionHandler.SendMessage(
+        GetBlockHeaders(Left(DaoBlockNumber), maxHeaders = 1, skip = 0, reverse = 0)
+      )
+      val waitingForDaoTimeout = system.scheduler.scheduleOnce(waitForStatusInterval, self, NoDaoHeaderReceived)
+      context become waitingForChainForkCheck(rlpxConnection, status, waitingForDaoTimeout)
+    case StatusReceiveTimeout =>
+      log.warning("Timeout while waiting for peer {} status", peerId)
+      disconnectFromPeer(rlpxConnection, Disconnect.Reasons.TimeoutOnReceivingAMessage)
+    case MessageReceived(d: Disconnect) =>
+      log.info("Received {} from peer {}. Closing connection", d, peerId)
+      context stop self
+  }
 
+  def waitingForChainForkCheck(rlpxConnection: ActorRef, status: Status, timeout: Cancellable): Receive = handleTerminated orElse
+    handlePing(rlpxConnection) orElse {
+    case MessageReceived(msg@BlockHeaders(blockHeader +: Nil)) if blockHeader.number == DaoBlockNumber =>
+      timeout.cancel()
+      log.info("DAO Fork header received from peer {} - {}", peerId, Hex.toHexString(blockHeader.hash))
+      if (daoForkValidator.validate(msg).isEmpty) {
+        log.warning("Peer {} is running the ETC chain", peerId)
         context become new HandshakedHandler(rlpxConnection).receive
       } else {
-        log.info("Peer does not support eth {} protocol. Disconnecting.", Message.PV63.toByte)
-        rlpxConnection ! SendMessage(Disconnect(Disconnect.Reasons.IncompatibleP2pProtocolVersion))
-        context.system.scheduler.scheduleOnce(5.seconds, self, PoisonPill)
+        log.warning("Peer {} is not running the ETC fork, disconnecting", peerId)
+        disconnectFromPeer(rlpxConnection, Disconnect.Reasons.UselessPeer)
+      }
+    case MessageReceived(d: Disconnect) =>
+      log.info("Received {} from peer {}. Closing connection", d, peerId)
+      disconnectFromPeer(rlpxConnection, Disconnect.Reasons.UselessPeer)
+    case NoDaoHeaderReceived =>
+      // FIXME We need to do some checking related to our blockchain. If we haven't arrived to the DAO block we might
+      // take advantage of this peer and grab as much blocks as we can until DAO.
+      // ATM we will only check by DaoBlockTotalDifficulty
+      if (status.totalDifficulty < DaoBlockTotalDifficulty) {
+        log.info("Peer {} probably hasn't arrived to DAO yet, so we will wait")
+        context become new HandshakedHandler(rlpxConnection).receive
       }
+      else {
+        log.info("Peer {} probably hasn't arrived to DAO yet, so we will wait")
+        disconnectFromPeer(rlpxConnection, Disconnect.Reasons.UselessPeer)
+      }
+  }
+
+  private def disconnectFromPeer(rlpxConnection: ActorRef, reason: Int): Unit = {
+    rlpxConnection ! SendMessage(Disconnect(Disconnect.Reasons.TimeoutOnReceivingAMessage))
```

Comment:
I believe it should say `RLPxConnection.SendMessage`, ATM we have two distinct messages (one in PeerActor and one in RLPxConn) we could think about having just one such msg